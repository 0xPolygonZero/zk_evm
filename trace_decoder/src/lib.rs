#![feature(linked_list_cursors)]
#![feature(trait_alias)]
#![feature(iter_array_chunks)]
// TODO: address these lints
#![allow(unused)]
#![allow(private_interfaces)]

mod compact;
pub mod decoding;
mod deserializers;
pub mod processed_block_trace;
pub mod trace_protocol;
pub mod types;
pub mod utils;
