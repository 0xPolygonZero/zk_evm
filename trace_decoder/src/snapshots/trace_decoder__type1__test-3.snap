---
source: trace_decoder/src/lib.rs
expression: reshaped
---
Reshape {
    state: HashedPartialTrie {
        node: Branch {
            children: [
                HashedPartialTrie {
                    node: Leaf {
                        nibbles: Nibbles {
                            count: 63,
                            packed: "0x3601462093b5945d1676df093446790fd31b20e7b12a2e8e5e09d068109616b",
                        },
                        value: [
                            248,
                            78,
                            128,
                            138,
                            2,
                            30,
                            25,
                            224,
                            201,
                            186,
                            178,
                            64,
                            0,
                            0,
                            160,
                            86,
                            232,
                            31,
                            23,
                            27,
                            204,
                            85,
                            166,
                            255,
                            131,
                            69,
                            230,
                            146,
                            192,
                            248,
                            110,
                            91,
                            72,
                            224,
                            27,
                            153,
                            108,
                            173,
                            192,
                            1,
                            98,
                            47,
                            181,
                            227,
                            99,
                            180,
                            33,
                            160,
                            197,
                            210,
                            70,
                            1,
                            134,
                            247,
                            35,
                            60,
                            146,
                            126,
                            125,
                            178,
                            220,
                            199,
                            3,
                            192,
                            229,
                            0,
                            182,
                            83,
                            202,
                            130,
                            39,
                            59,
                            123,
                            250,
                            216,
                            4,
                            93,
                            133,
                            164,
                            112,
                        ],
                    },
                    hash: RwLock {
                        data: None,
                    },
                },
                HashedPartialTrie {
                    node: Leaf {
                        nibbles: Nibbles {
                            count: 63,
                            packed: "0x468288056310c82aa4c01a7e12a10f8111a0560e72b700555479031b86c357d",
                        },
                        value: [
                            248,
                            68,
                            128,
                            1,
                            160,
                            86,
                            232,
                            31,
                            23,
                            27,
                            204,
                            85,
                            166,
                            255,
                            131,
                            69,
                            230,
                            146,
                            192,
                            248,
                            110,
                            91,
                            72,
                            224,
                            27,
                            153,
                            108,
                            173,
                            192,
                            1,
                            98,
                            47,
                            181,
                            227,
                            99,
                            180,
                            33,
                            160,
                            197,
                            210,
                            70,
                            1,
                            134,
                            247,
                            35,
                            60,
                            146,
                            126,
                            125,
                            178,
                            220,
                            199,
                            3,
                            192,
                            229,
                            0,
                            182,
                            83,
                            202,
                            130,
                            39,
                            59,
                            123,
                            250,
                            216,
                            4,
                            93,
                            133,
                            164,
                            112,
                        ],
                    },
                    hash: RwLock {
                        data: None,
                    },
                },
                HashedPartialTrie {
                    node: Empty,
                    hash: RwLock {
                        data: None,
                    },
                },
                HashedPartialTrie {
                    node: Empty,
                    hash: RwLock {
                        data: None,
                    },
                },
                HashedPartialTrie {
                    node: Hash(
                        0x1a697e814758281972fcd13bc9707dbcd2f195986b05463d7b78426508445a04,
                    ),
                    hash: RwLock {
                        data: None,
                    },
                },
                HashedPartialTrie {
                    node: Leaf {
                        nibbles: Nibbles {
                            count: 63,
                            packed: "0xb70e80538acdabd6137353b0f9d8d149f4dba91e8be2e7946e409bfdbe685b9",
                        },
                        value: [
                            248,
                            68,
                            128,
                            1,
                            160,
                            86,
                            232,
                            31,
                            23,
                            27,
                            204,
                            85,
                            166,
                            255,
                            131,
                            69,
                            230,
                            146,
                            192,
                            248,
                            110,
                            91,
                            72,
                            224,
                            27,
                            153,
                            108,
                            173,
                            192,
                            1,
                            98,
                            47,
                            181,
                            227,
                            99,
                            180,
                            33,
                            160,
                            197,
                            210,
                            70,
                            1,
                            134,
                            247,
                            35,
                            60,
                            146,
                            126,
                            125,
                            178,
                            220,
                            199,
                            3,
                            192,
                            229,
                            0,
                            182,
                            83,
                            202,
                            130,
                            39,
                            59,
                            123,
                            250,
                            216,
                            4,
                            93,
                            133,
                            164,
                            112,
                        ],
                    },
                    hash: RwLock {
                        data: None,
                    },
                },
                HashedPartialTrie {
                    node: Leaf {
                        nibbles: Nibbles {
                            count: 63,
                            packed: "0x89802d6ed1a28b049e9d4fe5334c5902fd9bc00c42821c82f82ee2da10be908",
                        },
                        value: [
                            248,
                            68,
                            128,
                            1,
                            160,
                            86,
                            232,
                            31,
                            23,
                            27,
                            204,
                            85,
                            166,
                            255,
                            131,
                            69,
                            230,
                            146,
                            192,
                            248,
                            110,
                            91,
                            72,
                            224,
                            27,
                            153,
                            108,
                            173,
                            192,
                            1,
                            98,
                            47,
                            181,
                            227,
                            99,
                            180,
                            33,
                            160,
                            197,
                            210,
                            70,
                            1,
                            134,
                            247,
                            35,
                            60,
                            146,
                            126,
                            125,
                            178,
                            220,
                            199,
                            3,
                            192,
                            229,
                            0,
                            182,
                            83,
                            202,
                            130,
                            39,
                            59,
                            123,
                            250,
                            216,
                            4,
                            93,
                            133,
                            164,
                            112,
                        ],
                    },
                    hash: RwLock {
                        data: None,
                    },
                },
                HashedPartialTrie {
                    node: Empty,
                    hash: RwLock {
                        data: None,
                    },
                },
                HashedPartialTrie {
                    node: Branch {
                        children: [
                            HashedPartialTrie {
                                node: Empty,
                                hash: RwLock {
                                    data: None,
                                },
                            },
                            HashedPartialTrie {
                                node: Empty,
                                hash: RwLock {
                                    data: None,
                                },
                            },
                            HashedPartialTrie {
                                node: Empty,
                                hash: RwLock {
                                    data: None,
                                },
                            },
                            HashedPartialTrie {
                                node: Empty,
                                hash: RwLock {
                                    data: None,
                                },
                            },
                            HashedPartialTrie {
                                node: Empty,
                                hash: RwLock {
                                    data: None,
                                },
                            },
                            HashedPartialTrie {
                                node: Leaf {
                                    nibbles: Nibbles {
                                        count: 62,
                                        packed: "0x56274a27dd7524955417c11ecd917251cc7c4c8310f4c7e4bd3c304d3d9a79",
                                    },
                                    value: [
                                        248,
                                        78,
                                        128,
                                        138,
                                        2,
                                        30,
                                        25,
                                        224,
                                        201,
                                        186,
                                        178,
                                        64,
                                        0,
                                        0,
                                        160,
                                        86,
                                        232,
                                        31,
                                        23,
                                        27,
                                        204,
                                        85,
                                        166,
                                        255,
                                        131,
                                        69,
                                        230,
                                        146,
                                        192,
                                        248,
                                        110,
                                        91,
                                        72,
                                        224,
                                        27,
                                        153,
                                        108,
                                        173,
                                        192,
                                        1,
                                        98,
                                        47,
                                        181,
                                        227,
                                        99,
                                        180,
                                        33,
                                        160,
                                        197,
                                        210,
                                        70,
                                        1,
                                        134,
                                        247,
                                        35,
                                        60,
                                        146,
                                        126,
                                        125,
                                        178,
                                        220,
                                        199,
                                        3,
                                        192,
                                        229,
                                        0,
                                        182,
                                        83,
                                        202,
                                        130,
                                        39,
                                        59,
                                        123,
                                        250,
                                        216,
                                        4,
                                        93,
                                        133,
                                        164,
                                        112,
                                    ],
                                },
                                hash: RwLock {
                                    data: None,
                                },
                            },
                            HashedPartialTrie {
                                node: Empty,
                                hash: RwLock {
                                    data: None,
                                },
                            },
                            HashedPartialTrie {
                                node: Empty,
                                hash: RwLock {
                                    data: None,
                                },
                            },
                            HashedPartialTrie {
                                node: Empty,
                                hash: RwLock {
                                    data: None,
                                },
                            },
                            HashedPartialTrie {
                                node: Empty,
                                hash: RwLock {
                                    data: None,
                                },
                            },
                            HashedPartialTrie {
                                node: Empty,
                                hash: RwLock {
                                    data: None,
                                },
                            },
                            HashedPartialTrie {
                                node: Empty,
                                hash: RwLock {
                                    data: None,
                                },
                            },
                            HashedPartialTrie {
                                node: Leaf {
                                    nibbles: Nibbles {
                                        count: 62,
                                        packed: "0x3ab0970b73895b8c9959bae685c3a19f45eb5ad89d42b52a340ec4ac204d19",
                                    },
                                    value: [
                                        248,
                                        68,
                                        128,
                                        1,
                                        160,
                                        86,
                                        232,
                                        31,
                                        23,
                                        27,
                                        204,
                                        85,
                                        166,
                                        255,
                                        131,
                                        69,
                                        230,
                                        146,
                                        192,
                                        248,
                                        110,
                                        91,
                                        72,
                                        224,
                                        27,
                                        153,
                                        108,
                                        173,
                                        192,
                                        1,
                                        98,
                                        47,
                                        181,
                                        227,
                                        99,
                                        180,
                                        33,
                                        160,
                                        197,
                                        210,
                                        70,
                                        1,
                                        134,
                                        247,
                                        35,
                                        60,
                                        146,
                                        126,
                                        125,
                                        178,
                                        220,
                                        199,
                                        3,
                                        192,
                                        229,
                                        0,
                                        182,
                                        83,
                                        202,
                                        130,
                                        39,
                                        59,
                                        123,
                                        250,
                                        216,
                                        4,
                                        93,
                                        133,
                                        164,
                                        112,
                                    ],
                                },
                                hash: RwLock {
                                    data: None,
                                },
                            },
                            HashedPartialTrie {
                                node: Empty,
                                hash: RwLock {
                                    data: None,
                                },
                            },
                            HashedPartialTrie {
                                node: Empty,
                                hash: RwLock {
                                    data: None,
                                },
                            },
                            HashedPartialTrie {
                                node: Empty,
                                hash: RwLock {
                                    data: None,
                                },
                            },
                        ],
                        value: [],
                    },
                    hash: RwLock {
                        data: None,
                    },
                },
                HashedPartialTrie {
                    node: Empty,
                    hash: RwLock {
                        data: None,
                    },
                },
                HashedPartialTrie {
                    node: Leaf {
                        nibbles: Nibbles {
                            count: 63,
                            packed: "0x876da518a393dbd067dc72abfa08d475ed6447fca96d92ec3f9e7eba503ca61",
                        },
                        value: [
                            248,
                            68,
                            128,
                            1,
                            160,
                            86,
                            232,
                            31,
                            23,
                            27,
                            204,
                            85,
                            166,
                            255,
                            131,
                            69,
                            230,
                            146,
                            192,
                            248,
                            110,
                            91,
                            72,
                            224,
                            27,
                            153,
                            108,
                            173,
                            192,
                            1,
                            98,
                            47,
                            181,
                            227,
                            99,
                            180,
                            33,
                            160,
                            197,
                            210,
                            70,
                            1,
                            134,
                            247,
                            35,
                            60,
                            146,
                            126,
                            125,
                            178,
                            220,
                            199,
                            3,
                            192,
                            229,
                            0,
                            182,
                            83,
                            202,
                            130,
                            39,
                            59,
                            123,
                            250,
                            216,
                            4,
                            93,
                            133,
                            164,
                            112,
                        ],
                    },
                    hash: RwLock {
                        data: None,
                    },
                },
                HashedPartialTrie {
                    node: Empty,
                    hash: RwLock {
                        data: None,
                    },
                },
                HashedPartialTrie {
                    node: Empty,
                    hash: RwLock {
                        data: None,
                    },
                },
                HashedPartialTrie {
                    node: Leaf {
                        nibbles: Nibbles {
                            count: 63,
                            packed: "0x52688a8f926c816ca1e079067caba944f158e764817b83fc43594370ca9cf62",
                        },
                        value: [
                            248,
                            68,
                            128,
                            1,
                            160,
                            86,
                            232,
                            31,
                            23,
                            27,
                            204,
                            85,
                            166,
                            255,
                            131,
                            69,
                            230,
                            146,
                            192,
                            248,
                            110,
                            91,
                            72,
                            224,
                            27,
                            153,
                            108,
                            173,
                            192,
                            1,
                            98,
                            47,
                            181,
                            227,
                            99,
                            180,
                            33,
                            160,
                            197,
                            210,
                            70,
                            1,
                            134,
                            247,
                            35,
                            60,
                            146,
                            126,
                            125,
                            178,
                            220,
                            199,
                            3,
                            192,
                            229,
                            0,
                            182,
                            83,
                            202,
                            130,
                            39,
                            59,
                            123,
                            250,
                            216,
                            4,
                            93,
                            133,
                            164,
                            112,
                        ],
                    },
                    hash: RwLock {
                        data: None,
                    },
                },
                HashedPartialTrie {
                    node: Leaf {
                        nibbles: Nibbles {
                            count: 63,
                            packed: "0x690b239ba3aaf993e443ae14aeffc44cf8d9931a79baed9fa141d0e4506e131",
                        },
                        value: [
                            248,
                            68,
                            128,
                            1,
                            160,
                            86,
                            232,
                            31,
                            23,
                            27,
                            204,
                            85,
                            166,
                            255,
                            131,
                            69,
                            230,
                            146,
                            192,
                            248,
                            110,
                            91,
                            72,
                            224,
                            27,
                            153,
                            108,
                            173,
                            192,
                            1,
                            98,
                            47,
                            181,
                            227,
                            99,
                            180,
                            33,
                            160,
                            197,
                            210,
                            70,
                            1,
                            134,
                            247,
                            35,
                            60,
                            146,
                            126,
                            125,
                            178,
                            220,
                            199,
                            3,
                            192,
                            229,
                            0,
                            182,
                            83,
                            202,
                            130,
                            39,
                            59,
                            123,
                            250,
                            216,
                            4,
                            93,
                            133,
                            164,
                            112,
                        ],
                    },
                    hash: RwLock {
                        data: None,
                    },
                },
                HashedPartialTrie {
                    node: Empty,
                    hash: RwLock {
                        data: None,
                    },
                },
            ],
            value: [],
        },
        hash: RwLock {
            data: None,
        },
    },
    code: {},
    storage: {
        0x03601462093b5945d1676df093446790fd31b20e7b12a2e8e5e09d068109616b: HashedPartialTrie {
            node: Empty,
            hash: RwLock {
                data: None,
            },
        },
        0x1468288056310c82aa4c01a7e12a10f8111a0560e72b700555479031b86c357d: HashedPartialTrie {
            node: Empty,
            hash: RwLock {
                data: None,
            },
        },
        0x5b70e80538acdabd6137353b0f9d8d149f4dba91e8be2e7946e409bfdbe685b9: HashedPartialTrie {
            node: Empty,
            hash: RwLock {
                data: None,
            },
        },
        0x689802d6ed1a28b049e9d4fe5334c5902fd9bc00c42821c82f82ee2da10be908: HashedPartialTrie {
            node: Empty,
            hash: RwLock {
                data: None,
            },
        },
        0x8556274a27dd7524955417c11ecd917251cc7c4c8310f4c7e4bd3c304d3d9a79: HashedPartialTrie {
            node: Empty,
            hash: RwLock {
                data: None,
            },
        },
        0x8c3ab0970b73895b8c9959bae685c3a19f45eb5ad89d42b52a340ec4ac204d19: HashedPartialTrie {
            node: Empty,
            hash: RwLock {
                data: None,
            },
        },
        0xa876da518a393dbd067dc72abfa08d475ed6447fca96d92ec3f9e7eba503ca61: HashedPartialTrie {
            node: Empty,
            hash: RwLock {
                data: None,
            },
        },
        0xd52688a8f926c816ca1e079067caba944f158e764817b83fc43594370ca9cf62: HashedPartialTrie {
            node: Empty,
            hash: RwLock {
                data: None,
            },
        },
        0xe690b239ba3aaf993e443ae14aeffc44cf8d9931a79baed9fa141d0e4506e131: HashedPartialTrie {
            node: Empty,
            hash: RwLock {
                data: None,
            },
        },
    },
}
